cpp/ql/src/semmle/code/cpp/dataflow/internal/DataFlowImpl.qll

```diff
@@ -1060,8 +1060,8 @@ private module LocalFlowBigStep {
       jumpStep(_, node, config) or
       additionalJumpStep(_, node, config) or
       node instanceof ParameterNode or
-      node instanceof OutNode or
-      node instanceof PostUpdateNode or
```

Comment:
We can't remove this line.  Taint steps that target array writes use `PostUpdateNode`s so there are more cases than just those associated with arguments and stores.

Reply:
(In the Java modelling of array taint, that is)

Reply:
Could you maybe add the following Java taint flow test:
```
foo() {
  int[] xs = new int[2];
  bar(xs);
  sink(xs);
}
bar(int[] xs) {
  xs[0] = taint();
}
```
and check that this line in fact makes the difference between flow or not.

Reply:
Sorry, no wait, that test won't actually do the trick.  I think the simpler
```
foo(int[] xs) {
  xs[0] = taint();
  sink(xs);
}
```
ought to highlight the issue.

Reply:
Hmm, actually, probably no again. Writing counter examples is tricky.

Reply:
Ignore all of the above, I've convinced myself that there's no problem here.